cmd/opampsupervisor/supervisor/healthchecker/healthchecker.go

```diff
@@ -0,0 +1,43 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package healthchecker
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"time"
+)
+
+type HTTPHealthChecker struct {
+	endpoint string
+}
+
+func NewHTTPHealthChecker(endpoint string) *HTTPHealthChecker {
+	return &HTTPHealthChecker{
+		endpoint: endpoint,
+	}
+}
+
+func (h *HTTPHealthChecker) Check(ctx context.Context) error {
+	req, err := http.NewRequestWithContext(ctx, "GET", h.endpoint, nil)
```

Comment:
This is probably a bit too immature and we should look at using confighttp.HTTPClientSettings to set this up, in a later PR

Reply:
That's a fair point. I've opened https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/23344 to address this and linked to it in this file.